src/test/java/org/openmrs/addonindex/domain/AddOnInfoAndVersionsTest.java

```diff
@@ -16,29 +17,43 @@
 @RunWith(SpringRunner.class)
 @JsonTest
 public class AddOnInfoAndVersionsTest {
-	
-	@Autowired
-	private ObjectMapper mapper;
-	
-	@Autowired
-	private JacksonTester<AddOnInfoAndVersions> json;
-	
-	@Test
-	public void testSerialization() throws Exception {
-		AddOnVersion version = new AddOnVersion();
-		version.setVersion(new Version("1.0"));
-		
-		AddOnInfoAndVersions info = new AddOnInfoAndVersions();
-		info.setName("OpenMRS");
-		info.setDescription("Write code, save lives");
-		info.addVersion(version);
-		
-		String json = mapper.writeValueAsString(info);
-		AddOnInfoAndVersions parsed = mapper.readValue(json, AddOnInfoAndVersions.class);
-		assertThat(parsed.getName(), is("OpenMRS"));
-		assertThat(parsed.getDescription(), is("Write code, save lives"));
-		assertThat(parsed.getVersions().size(), is(1));
-		assertThat(parsed.getVersions().get(0).getVersion().toString(), is("1.0"));
-	}
-	
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private JacksonTester<AddOnInfoAndVersions> json;
+
+    @Test
+    public void testSerialization() throws Exception {
+        AddOnVersion version = new AddOnVersion();
+        version.setVersion(new Version("1.0"));
+
+        AddOnInfoAndVersions info = new AddOnInfoAndVersions();
+        info.setName("OpenMRS");
+        info.setDescription("Write code, save lives");
+        info.addVersion(version);
+
+        String json = mapper.writeValueAsString(info);
+        AddOnInfoAndVersions parsed = mapper.readValue(json, AddOnInfoAndVersions.class);
+        assertThat(parsed.getName(), is("OpenMRS"));
+        assertThat(parsed.getDescription(), is("Write code, save lives"));
+        assertThat(parsed.getVersions().size(), is(1));
+        assertThat(parsed.getVersions().get(0).getVersion().toString(), is("1.0"));
+    }
+
+    @Test
+    public void testGetLatestMatchedVersion() throws Exception {
```

Comment:
In _this_ test class you should have 3 tests for different scenarios:

1. the latest version matches
2. some earlier version matches
3. no version matches